### PR TITLE
refactor: change counterparty structure.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

can also tweak any names as needed in this PR, colin already raised some naming alts in the other PR.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/packetserver/v1/tx.proto [hunk 1]
```diff
@@ -25,8 +25,8 @@ message MsgProvideCounterparty {
 
   option (gogoproto.goproto_getters) = false;
 
-  // client unique identifier
-  string client_id = 1;
+  // unique identifier we will use to write all packet messages sent to counterparty
+  string channel_id = 1;
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply proto/ibc/core/packetserver/v1/tx.proto:29]: what about `source_id`?

File: modules/core/keeper/msg_server.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (k *Keeper) IBCSoftwareUpgrade(goCtx context.Context, msg *clienttypes.MsgI
 func (k *Keeper) ProvideCounterparty(goCtx context.Context, msg *packetservertypes.MsgProvideCounterparty) (*packetservertypes.MsgProvideCounterpartyResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	creator, found := k.ClientKeeper.GetCreator(ctx, msg.ClientId)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/keeper/msg_server.go:151]: maybe its confusing intertwine terminology used for module permissions and client creation. 

The way I thought about it was: `msg.Signer` creates a client, thus they have the authority to bind a counterparty to it.

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/keeper/msg_server.go:151]: i.e. `SetClientAuthority(ctx, clientID, msg.Signer)` && `GetClientAuthority(ctx, clientID)`

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/keeper/msg_server.go:151]: I'm not quite sure how strong my pref is! 😆

File: proto/ibc/core/packetserver/v1/counterparty.proto [hunk 3]
```diff
@@ -9,8 +9,12 @@ import "ibc/core/commitment/v2/commitment.proto";
 
 // Counterparty defines the counterparty for a light client to implement IBC eureka protocol
 message Counterparty {
-  // the client identifier of the counterparty chain
+  // the client identifier of the light client representing the counterparty chain
   string client_id = 1;
-  // the merkle path that all ICS24 paths will be stored under
-  ibc.core.commitment.v2.MerklePath merkle_path_prefix = 2 [(gogoproto.nullable) = false];
+  // the counterparty identifier that must be used by the packet
+  string counterparty_channel = 2;
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply proto/ibc/core/packetserver/v1/counterparty.proto:15]: would definitely make sure any ids are suffixed with `_id` anyways

**Review by AdityaSripal — APPROVED** [review]
Thanks Jim! We can continue finalizing naming in the subsequent PRs

File: proto/ibc/core/packetserver/v1/tx.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply proto/ibc/core/packetserver/v1/tx.proto:29]: Yeah, that makes sense

File: proto/ibc/core/packetserver/v1/counterparty.proto [hunk 4]
```diff
@@ -9,8 +9,12 @@ import "ibc/core/commitment/v2/commitment.proto";
...
+  // the key path used to store packet flow messages that the counterparty
+  // will use to send to us. In backwards compatible cases, we will append the channelID and sequence in order to create
+  // the final path.
+  ibc.core.commitment.v2.MerklePath merkle_path_prefix = 3 [(gogoproto.nullable) = false];
```
Comment (damiannolan) [inline proto/ibc/core/packetserver/v1/counterparty.proto:19]: Was there a reason that `MerklePrefix` wasn't used here? cc @AdityaSripal

**Review by damiannolan — APPROVED** [review]
LGTM, agree with following up on namings as needs be

**Pull request merged**